probe-rs/src/architecture/arm/core/exception_handling/armv6m_armv7m_shared.rs

```diff
@@ -45,15 +65,130 @@ bitfield! {
 ) -> Result<Option<ExceptionInfo>, Error> {
+    let frame_return_address = get_stack_frame_return_address(stackframe_registers)?;
+
+    if ExcReturn(frame_return_address).is_exception_flag() == 0xF {
+        // This is an exception frame.
+        let raw_exception = exception_interface.raw_exception(stackframe_registers)?;
+        let registers = exception_interface.calling_frame_registers(
+            memory_interface,
+            stackframe_registers,
+            raw_exception,
+        )?;
+        let description =
+            exception_interface.exception_description(raw_exception, memory_interface)?;
+
+        let exception_frame_pc =
+            registers.get_register_value_by_role(&RegisterRole::ProgramCounter)?;
+
+        let mut handler_frame = StackFrame {
+            id: get_object_reference(),
+            function_name: description.clone(),
+            source_location: None,
+            registers,
+            pc: RegisterValue::U32(exception_frame_pc as u32),
+            frame_base: None,
+            is_inlined: false,
+            local_variables: None,
+            canonical_frame_address: None,
+        };
+
+        // A fault that is escalated to the priority of a HardFault retains the program counter value of the original fault,
+        // so we have to unwind the frame pointer that matches.
+        if raw_exception == 3 {
+            // A fault that is escalated to the priority of a HardFault retains program counter value of the original fault,
+            // So we have to unwind the frame pointer that matches.
+            // Determining the frame base may need the CFA (Canonical Frame Address) to be calculated first.
+            let mut unwind_context = Box::new(gimli::UnwindContext::new());
+            // let exception_interface = exception_handler_for_core(memory_interface.core.core_type());
+            // let instruction_set = memory_interface.core.instruction_set().ok();
+            let unwind_info = get_unwind_info(
+                &mut unwind_context,
+                &debug_info.frame_section,
+                exception_frame_pc,
+            )
+            .map_err(|error: DebugError| Error::Other(error.into()))?;
+            handler_frame.canonical_frame_address =
+                determine_cfa(&handler_frame.registers, unwind_info)?;
+            let Ok((_, functions)) = debug_info.get_function_dies(exception_frame_pc) else {
+                handler_frame.function_name = format!("{} : ERROR: No function information for the program counter ({exception_frame_pc:#010x}) that caused the exception.", handler_frame.function_name);
+                // Return the available exception info, along with the error information captured in the function_name.
+                return Ok(Some(ExceptionInfo {
+                    raw_exception,
+                    description,
+                    handler_frame,
+                }));
+            };
+            handler_frame.frame_base = functions[0]
```

Comment:
Can this not panic?

Reply:
I think you're right. Good catch!

Reply:
I found another instance in older code that had the same issue, and added a check for it.

In future, I will see if it makes sense to change the interface to NOT return an `OK(<empty_vec>)`